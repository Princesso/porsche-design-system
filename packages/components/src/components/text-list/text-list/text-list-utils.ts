import type { TextListItem } from '../text-list-item/text-list-item';
import type { TextListItemInternalHTMLProps } from '../text-list-item/text-list-item-utils';
import { forceUpdate } from '@stencil/core';

export const LIST_TYPES = ['unordered', 'ordered'] as const;
export type ListType = typeof LIST_TYPES[number];

export const ORDER_TYPES = ['numbered', 'alphabetically'] as const;
export type OrderType = typeof ORDER_TYPES[number];

export const syncTextListItemsProps = (host: HTMLElement, listType: ListType, orderType: OrderType): void => {
  Array.from(host.children).forEach((item: HTMLElement & TextListItem & TextListItemInternalHTMLProps) => {
    item.listType = listType;
    item.orderType = orderType;
    forceUpdate(item);
  });
};
