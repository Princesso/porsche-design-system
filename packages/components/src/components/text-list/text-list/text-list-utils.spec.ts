import { isListTypeOrdered, isOrderTypeNumbered } from './text-list-utils';

describe('isListTypeOrdered()', () => {
  it.each<Parameters<typeof isListTypeOrdered>>([['ordered'], ['unordered']])(
    'should return correct boolean for list type: %s',
    (...args) => {
      expect(isListTypeOrdered(...args)).toMatchSnapshot();
    }
  );
});

describe('isOrderTypeNumbered()', () => {
  it.each<Parameters<typeof isOrderTypeNumbered>>([['numbered'], ['alphabetically']])(
    'should return correct boolean for order type: %s',
    (...args) => {
      expect(isOrderTypeNumbered(...args)).toMatchSnapshot();
    }
  );
});
