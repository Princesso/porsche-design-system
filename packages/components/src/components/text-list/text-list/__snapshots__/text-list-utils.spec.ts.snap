// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isListTypeOrdered() should return correct boolean for list type: ordered 1`] = `true`;

exports[`isListTypeOrdered() should return correct boolean for list type: unordered 1`] = `false`;

exports[`isOrderTypeNumbered() should return correct boolean for order type: alphabetically 1`] = `false`;

exports[`isOrderTypeNumbered() should return correct boolean for order type: numbered 1`] = `true`;
